src/utils/useWeb3Connect.js

```diff
@@ -385,16 +400,14 @@ function useWeb3Connect() {
       return 0;
     }
     let balance = new web3ReadOnly.utils.BN(
-      await adaiContractReadOnly.methods.balanceOf(WHOOP_ADDRESS).call()
+      await adaiContractReadOnly.methods.balanceOf(DEPOSIT_ADDRESS).call()
     );
     let depositedAmount = new web3ReadOnly.utils.BN(
-      await daoContractReadOnly.methods.totalDepositedDai().call()
+      await depositContractReadOnly.methods.totalDepositedDai().call()
     );
-    // console.log({ balance });
-    let interest = balance - depositedAmount;
+    let interest = balance.sub(depositedAmount);
 
-    // let interestBn = new BN(interest);
-    return Number(web3ReadOnly.utils.fromWei('' + interest, 'ether'));
```

Comment:
Forcing this value into a string was causing problems because when it was very very small it was something that this couldn't handle. Anyway, using `.toString`  explicitly should be preferred to coercing it into a string like that.

fromWei is totally happy to deal with a BN directly.